### ISSUE TITLE
cmake regex error

### ISSUE THREAD

**Issue by Gibbz** [LaurentGomila/qt-android-cmake#30]
I get the following error, and so the script fails....

> Detected Android SDK build tools version 0.0.0
> CMake Error at android/AddQtAndroidApk.cmake:161 (string):
>   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
>   command.
> Call Stack (most recent call first):
>   CMakeLists.txt:100 (add_qt_android_apk)

This looks like its looking in the toolchain folder which for me has the following:

> aarch64-linux-android-4.9
> arm-linux-androideabi-4.9
> llvm
> renderscript
> x86-4.9

**Comment by Gibbz** [comment]
My paths are:

> export ANDROID_NDK="$PATH:/opt/android-ndk/"
> export ANDROID_SDK="$PATH:/opt/android-sdk/"
> export JAVA_HOME="$PATH:/usr/lib/jvm/default/"

Any idea how to fix this issue so I can get my app compiling?

Thanks!

**Comment by LaurentGomila** [comment]
I have no idea, sorry. You can try to start investigating the issue, and give more information (figure out what the script tries to do, log the relevant variables, ...).

**Comment by Gibbz** [comment]
Ive added a few lines to fix the issue:

```
    # Bronson - added these lines to pickup the ndk and toolchain root
    set(ANDROID_NDK ${QT_ANDROID_NDK_ROOT})
    set(ANDROID_TOOLCHAIN_ROOT ${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64)
```

The last problem is... What installs androiddeployqt?
`/usr/bin/androiddeployqt: Command not found`

**Comment by LaurentGomila** [comment]
> The last problem is... What installs androiddeployqt?
`/usr/bin/androiddeployqt: Command not found`

It should come with the Android build of Qt, in the "bin" directory (where qmake is).

**Comment by LaurentGomila** [comment]
Probably fixed with #25, feel free to re-open this issue if not.
